dev/NavigationView/NavigationView.cpp

```diff
@@ -3611,9 +3614,16 @@ void NavigationView::UpdatePaneShadow()
             }
         }
 
+
         // Shadow will get clipped if casting on the splitView.Content directly
         // Creating a canvas with negative margins as receiver to allow shadow to be drawn outside the content grid 
         winrt::Thickness shadowReceiverMargin = { -CompactPaneLength(), -c_paneElevationTranslationZ, -c_paneElevationTranslationZ, -c_paneElevationTranslationZ };
+
+        // Ensuring shadow is aligned to the left
+        shadowReceiver.HorizontalAlignment(winrt::HorizontalAlignment::Left);
+
+        // Ensure shadow is as wide as the pane when it is open
+        shadowReceiver.Width(OpenPaneLength() + c_paneElevationTranslationZ);
```

Comment:
Should it be 2*c_paneElevationTranslationZ since there are negative margins on both left and right?

Reply:
Good that you thought about that, I totally missed that. However, since the shadow does not exceed outside to the left of the NavigationView (afaik), we don't need that.

Note that in the following screenshot, the NavigationView has a margin of 50 and the red border represents the shadow canvas.
![navviewshadow-bug](https://user-images.githubusercontent.com/16122379/66078628-a76cec80-e562-11e9-8230-c58a5efc4bb2.PNG)

Setting the width of the shadow canvas to a larger value only makes it expand to the right.

Reply:
I see. Now I think the problem comes from the left negative margin.

![Annotation 2019-10-02 143230](https://user-images.githubusercontent.com/4424330/66083734-cbf4a480-e521-11e9-95db-505cdcc56445.png)

The pane is aligned to the left edge of app window. Seems like adding a left negative margin in this case would decrease the content width instead of taking up more space on the left. I think we should remove the -CompactPaneLength() left margin, then shadow receiver width can just be OpenPaneLength(), top and right margin should do the work to make sure there's enough space for shadow.

![Annotation 2019-10-02 144518](https://user-images.githubusercontent.com/4424330/66084381-5ee20e80-e523-11e9-83a4-05031f4a7daf.png)

Reply:
Should be fine now with latest commit.